FreeRTOS_ARP.c

```diff
@@ -135,83 +135,93 @@ eFrameProcessingResult_t eARPProcessPacket( NetworkBufferDescriptor_t * pxNetwor
+                                         ( unsigned ) FreeRTOS_ntohl( ulTargetProtocolAddress ),
+                                         ( unsigned ) ( FreeRTOS_ntohl( ( pxTargetEndPoint != NULL ) ? pxTargetEndPoint->ipv4_settings.ulIPAddress : 0U ) ) ) );
+            }
+        #endif /* ( ipconfigHAS_PRINTF != 0 ) */
 
-    /* ulTargetProtocolAddress won't be used unless logging is enabled. */
-    ( void ) ulTargetProtocolAddress;
+        /* ulTargetProtocolAddress won't be used unless logging is enabled. */
+        ( void ) ulTargetProtocolAddress;
 
-    if( ( pxTargetEndPoint != NULL ) && ( pxTargetEndPoint->bits.bEndPointUp != pdFALSE_UNSIGNED ) )
-    {
-        switch( pxARPHeader->usOperation )
+        if( ( pxTargetEndPoint != NULL ) && ( pxTargetEndPoint->bits.bEndPointUp != pdFALSE_UNSIGNED ) )
         {
-            case ipARP_REQUEST:
-                vARPProcessPacketRequest( pxARPFrame, pxTargetEndPoint, ulSenderProtocolAddress );
-                eReturn = eReturnEthernetFrame;
-                break;
-
-            case ipARP_REPLY:
-                vARPProcessPacketReply( pxARPFrame, pxTargetEndPoint, ulSenderProtocolAddress );
-                break;
-
-            default:
-                /* Invalid. */
-                break;
+            switch( pxARPHeader->usOperation )
+            {
+                case ipARP_REQUEST:
+                    vARPProcessPacketRequest( pxARPFrame, pxTargetEndPoint, ulSenderProtocolAddress );
+                    eReturn = eReturnEthernetFrame;
+                    break;
+
+                case ipARP_REPLY:
+                    vARPProcessPacketReply( pxARPFrame, pxTargetEndPoint, ulSenderProtocolAddress );
+                    break;
+
+                default:
+                    /* Invalid. */
+                    break;
+            }
         }
-    }
 
-    /* Process received ARP frame to see if there is a clash. */
-    #if ( ipconfigARP_USE_CLASH_DETECTION != 0 )
-        {
-            if( pxSourceEndPoint != NULL )
+        /* Process received ARP frame to see if there is a clash. */
+        #if ( ipconfigARP_USE_CLASH_DETECTION != 0 )
             {
-                xARPHadIPClash = pdTRUE;
-                /* Remember the MAC-address of the other device which has the same IP-address. */
-                ( void ) memcpy( xARPClashMacAddress.ucBytes, pxARPHeader->xSenderHardwareAddress.ucBytes, sizeof( xARPClashMacAddress.ucBytes ) );
+                if( pxSourceEndPoint != NULL )
+                {
+                    xARPHadIPClash = pdTRUE;
+                    /* Remember the MAC-address of the other device which has the same IP-address. */
+                    ( void ) memcpy( xARPClashMacAddress.ucBytes, pxARPHeader->xSenderHardwareAddress.ucBytes, sizeof( xARPClashMacAddress.ucBytes ) );
+                }
             }
-        }
-    #endif /* ipconfigARP_USE_CLASH_DETECTION */
+        #endif /* ipconfigARP_USE_CLASH_DETECTION */
+    }
```

Comment:
As I wrote in Chime, could you add a trace macro here, something like:
~~~c
if( good )
{
    /* Handle the packet. */
}
else
{
    iptraceDROPPED_INVALID_PACKET( pxARPHeader );
}
~~~

Reply:
Yes, I added the trace macro. Thank you.